### ISSUE TITLE
after/ftplugin/c.lua is not sourced

### ISSUE THREAD

**Issue by seblj** [neovim/neovim#14781]
- `nvim --version`: NVIM v0.5.0-dev+1396-g81a6b7088
- Operating system/version: Ubuntu 20.04
- Terminal name/version: ssh through iTerm Build 3.4.20210523-nightly
- `$TERM`: xterm-256color

### Steps to reproduce using

Build latest release with #14686 merged
Create file `after/ftplugin/c.lua`
Content of file: `print("sourced")`

Open a c file

### Actual behaviour

Doesn't print out anything

### Expected behaviour

Should print out `sourced` if I understood the PR correctly

### Additional information

Colorscheme and after/plugin seems to work

:scriptnames
```
  1: /usr/local/share/nvim/runtime/filetype.vim
  2: /usr/local/share/nvim/runtime/ftplugin.vim
  3: /usr/local/share/nvim/runtime/indent.vim
  4: /usr/local/share/nvim/runtime/syntax/syntax.vim
  5: /usr/local/share/nvim/runtime/syntax/synload.vim
  6: /usr/local/share/nvim/runtime/syntax/syncolor.vim
  7: /usr/local/share/nvim/runtime/plugin/gzip.vim
  8: /usr/local/share/nvim/runtime/plugin/health.vim
  9: /usr/local/share/nvim/runtime/plugin/man.vim
 10: /usr/local/share/nvim/runtime/plugin/matchit.vim
 11: /usr/local/share/nvim/runtime/pack/dist/opt/matchit/plugin/matchit.vim
 12: /usr/local/share/nvim/runtime/plugin/matchparen.vim
 13: /usr/local/share/nvim/runtime/plugin/netrwPlugin.vim
 14: /usr/local/share/nvim/runtime/plugin/rplugin.vim
 15: /usr/local/share/nvim/runtime/plugin/shada.vim
 16: /usr/local/share/nvim/runtime/plugin/spellfile.vim
 17: /usr/local/share/nvim/runtime/plugin/tarPlugin.vim
 18: /usr/local/share/nvim/runtime/plugin/tohtml.vim
 19: /usr/local/share/nvim/runtime/plugin/tutor.vim
 20: /usr/local/share/nvim/runtime/plugin/zipPlugin.vim
 21: /usr/local/share/nvim/runtime/autoload/dist/ft.vim
 22: /usr/local/share/nvim/runtime/ftplugin/c.vim
 23: /usr/local/share/nvim/runtime/indent/c.vim
 24: /usr/local/share/nvim/runtime/syntax/c.vim
```

EDIT: Now noticing scriptnames doesn't list the file even if I create a file `after/plugin/test.lua` with `print("sourcing")`. However, this is printing `sourcing` when I open a buffer

**Comment by seblj** [comment]
My bad, can't print inside file. It seems to be sourcing correctly!

**Comment by shadmansaleh** [comment]
> My bad, can't print inside file.

I'm not sure why but for me printed statements appear after entering insert mode . Maybe nvim has a buffer or delay of some sort .

> Now noticing scriptnames doesn't list the file

This should be resolved with #14788

**Comment by seblj** [comment]
For me it doesn't print at all, but it's not a big deal since it runs whatever options I set
